src/iosource/Packet.h

```diff
@@ -191,23 +193,35 @@ class Packet {
 	 * (Outermost) VLAN tag if any, else 0. Valid iff Layer2Valid()
 	 * returns true.
 	 */
-	uint32_t vlan;
+	uint16_t vlan;
```

Comment:
True that current usage (802.1q / 802.1ad) don't need 32-bits, but maybe other reason to leave be in case others do (or will) need more storage?  e.g. not sure if it makes sense for some encapsulations to set/reuse this field, but things like VXLAN, NVGRE, and possibly more seem to commonly be using 24-bit IDs.  And a quick googling around after seeing 802.1aq noted as supporting 16 million IDs seems like it truly does have a 24-bit field, and not (only) separate inner and outer like 802.1ad.  Again, unsure this change realistically breaks anything at the moment, but if it's going to take a while to research the justification, suggest removing from this patch since not related.